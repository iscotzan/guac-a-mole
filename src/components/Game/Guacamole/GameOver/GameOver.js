import * as React from "react";
import { Grid } from "semantic-ui-react";
import { Translate } from "react-localize-redux";
import SplashButton from "./../SplashButton/SplashButton";
const GameOver = props => {
  const showGameOver = props.gameOver ? "show-div textAndScores" : "hide-div";

  return (
    <div className="game-over-container centered container-style">
      {/* game over stats */}
      <Grid centered>
        <div className={showGameOver}>
          <h1>
            <Translate id="translations.gameOver">Game Over</Translate>
          </h1>
          <div>
            <h3>
              <Translate id="translations.yourScore">Your Score</Translate>:
              {props.score}
            </h3>
            <h5>
              <Translate id="translations.highScore">High Score</Translate>:
              {props.highScore < props.score ? props.score : props.highScore}
            </h5>
          </div>
        </div>
        <div>
          <SplashButton
            buttonText={<Translate id="translations.play">Play</Translate>}
            onClick={props.restartGame}
          />
        </div>
      </Grid>
    </div>
  );
};

export default GameOver;
