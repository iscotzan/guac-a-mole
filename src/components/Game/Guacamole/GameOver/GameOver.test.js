import React from "react";
import GameOver from "./GameOver";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<GameOver gameOver={true} score={7777} highScore={10000} />);
describe("<GameOver />", () => {
  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("renders without crashing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
