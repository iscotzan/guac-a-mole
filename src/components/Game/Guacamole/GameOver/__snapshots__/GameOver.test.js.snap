// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameOver /> renders without crashing 1`] = `
<div
  className="game-over-container centered container-style"
>
  <Grid
    centered={true}
  >
    <div
      className="show-div textAndScores"
    >
      <h1>
        <Translate
          id="translations.gameOver"
        >
          Game Over
        </Translate>
      </h1>
      <div>
        <h3>
          <Translate
            id="translations.yourScore"
          >
            Your Score
          </Translate>
          :
          7777
        </h3>
        <h5>
          <Translate
            id="translations.highScore"
          >
            High Score
          </Translate>
          :
          10000
        </h5>
      </div>
    </div>
    <div>
      <SplashButton
        buttonText={
          <Translate
            id="translations.play"
          >
            Play
          </Translate>
        }
      />
    </div>
  </Grid>
</div>
`;
